pkg/proxy/apis/config/scheme/testdata/KubeProxyConfiguration/after/v1alpha1.yaml

```diff
@@ -37,6 +37,13 @@ ipvs:
   tcpTimeout: 0s
   udpTimeout: 0s
 kind: KubeProxyConfiguration
+logging:
+  flushFrequency: 5000000000
```

Comment:
this seems like a bug; it should serialize to "`5s`", not "`5000000000`"

Reply:
After serialize, the default values for logging are indeed these, see https://github.com/kubernetes/kubernetes/blob/b248684334c50027feb20cf42bcc8c48e4f9ede4/pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/v1beta1.yaml#L54-L59

Reply:
This seems like it's a bug in the shared component-base code then. Ah, known bug, couldn't be fixed because of API concerns. Probably to be fixed with a new field later. #107618

Reply:
before we expand to using this in a new component, can we fix this up... a type that accepts reading either int time.Duration or string metav1.Duration would make this more usable without breaking reading existing config files

sketch in the last commit of https://github.com/liggitt/kubernetes/commits/flushType

Reply:
@liggitt: regarding your proposed type... I think I had suggested something like that and one of the concerns at the time was that tools which write the config will then do with the new encoding (string), while existing components will only support int. Should `MarshalJSON` perhaps encode as int to address that?

@cyclinder: can you perhaps cherry-pick that commit into this PR? Then we can comment on the code directly.


Wasn't one of the concerns about changing the default encoding of that field from int to string that

Reply:
@pohly sure, but cherry-pick that commit into this PR or open another PR for this, which is better?

Reply:
The effect is easier to see when cherry-picked into this PR, so I prefer doing that even if it is, strictly speaking, an unrelated change.

Keep it as a separate commit and we can still merge it separately later.